docs: Reshape Endpoints & APIs concepts, archive deprecated streaming (#673)
serverpod/serverpod_docs: docs/06-concepts/01-working-with-endpoints/02-endpoint-inheritance.md

---

> description: Base endpoints on other endpoints with inheritance, override behavior from Serverpod modules, and control which subclasses generate client code.

Reviewer: Non-blocking: lead the `description` with the topic noun.

```suggestion
description: Endpoint inheritance lets one endpoint extend another, override behavior from Serverpod modules, and control which subclasses generate client code.
```
